File: EcoJardimPro - BackEnd/EcoJardimProject/EcoJardimProject.Domain/Entities/Cliente.cs
using EcoJardimProject.Domain.Base;

namespace EcoJardimProject.Domain.Entities;

public class Cliente : IEntity
{

    public long Id { get; set; }
    public string Nome { get; set; }
    public string Sobrenome { get; set; }
    public string Documento { get; set; }
    public Endereco? Endereco { get; set; }
    public Contato? Contato { get; set; }
    public ICollection<Projeto> Projetos { get; set; }
    public Cliente()
    {
        Projetos = new List<Projeto>();
    }
    
    public List<string> Validate()
    {
        var errors = new List<string>();

        if (string.IsNullOrEmpty(Nome))
        {
            errors.Add("Nome é obrigatório");
        }

        if (string.IsNullOrEmpty(Sobrenome))
        {
            errors.Add("Sobrenome é obrigatório");
        }
        return errors;
    }
}

File: EcoJardimPro - FronEnd/ecojardimprofrontend/src/models/cliente/cliente.ts
import Contato from "../objetosValor/Contato";
import Endereco from "../objetosValor/Endereco";
import Projeto from "../projeto/Projeto";

export default class Cliente{
    public id: number;
    public nome: string;
    public sobrenome: string;
    public documento: string;
    public endereco!: Endereco;
    public contato!: Contato;
    public projetos: Projeto[] = [];

    constructor() {
        this.id = 0
        this.nome = ''
        this.sobrenome = ''
        this.documento = ''
        this.endereco = new Endereco();
        this.contato = new Contato();
        this.projetos = []
    }
}